packages/integration-tests/src/test-e2e/create_amplify.test.ts

```diff
@@ -115,11 +115,6 @@ void describe('create-amplify script', { concurrency: concurrency }, () => {
     );
   });
 
-  after(async () => {
-    // stop the npm proxy
-    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
```

Comment:
Don't stop `npm-proxy` because `deploy.test.ts` will use it.

Reply:
deploy.test.ts shouldn't depend on setup from create_amplify. Each test suite should handle it's own setup and teardown. If the setup and teardown of two different test suites conflicts with each other, then we either need to run those suites serially, or figure out some other strategy for doing the setup without introducing coupling between tests.

Reply:
Yea, I'm thinking if we should publish locally in the GH workflow for all the e2e tests.
Setup test for each of them would be too time-consuming and expensive. 

What do you think?

Reply:
I was thinking about this problem before when I was working on initial version of canary tests and I wanted to include them in normal e2e test run too. I ended up not solving this because canaries never got included in normal e2e tests.

What we need here is to develop component that manages npm-proxy lifecycle using OS primitives or file system. The reason for this that npm test runner runs each test file as separate Node process.
On the high level what that component should do is (assuming we us fs):
1. Attempt to start npm proxy or skip depending on state/lock recorded on file system (I'd use something like https://en.wikipedia.org/wiki/Double-checked_locking for that)
2. When handling test start/tear down this should somehow keep record and stop npm proxy when last test tears down.
3. This should be somehow resilient to inconsistent state (in case tests crash) or it should be easy to notice it and recover.

I think this piece alone deserves separate PR (which we could do on `main`).

Reply:
One other option to solve this particular problem is to wrap node test runner cli with custom script and use programmatic api https://nodejs.org/api/test.html#runoptions . This should allow adding "before/after all test suites" logic.